src/test/java/io/pinecone/PineconeIndexOperationClientTest.java

```diff
@@ -53,12 +58,113 @@ public void testIndexCreation() throws IOException {
+        when(mockBoundRequestBuilder.setHeader(anyString(), anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.setBody(anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.execute()).thenReturn(mockResponseFuture);
+        when(mockResponseFuture.toCompletableFuture()).thenReturn(mockCompletableFuture);
+
+        PineconeIndexOperationClient indexDeletionService = new PineconeIndexOperationClient(clientConfig, mockClient);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest().indexName("test_name");
+
+        try{
+            indexDeletionService.createIndex(createIndexRequest);
+            fail("Expected validation exception not occurred");
+        }
+        catch (PineconeValidationException validationException) {
+            System.out.println("Expected PineconeValidationException with dimension not null occurred!");
+        }
+        catch (IOException e) {
+
+        }
+    }
+
+    @Test
+    public void testIndexCreationWithAllFields() {
+        AsyncHttpClient mockClient = mock(DefaultAsyncHttpClient.class);
+        PineconeClientConfig clientConfig = new PineconeClientConfig().withApiKey("testApiKey").withEnvironment("testEnvironment");
+        Response mockResponse = mock(Response.class);
+        BoundRequestBuilder mockBoundRequestBuilder = mock(BoundRequestBuilder.class);
+        ListenableFuture<Response> mockResponseFuture = mock(ListenableFuture.class);
+        CompletableFuture<Response> mockCompletableFuture = mock(CompletableFuture.class);
+
+        when(mockResponse.getStatusCode()).thenReturn(201);
+        when(mockClient.prepare(anyString(), anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.setHeader(anyString(), anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.setBody(anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.execute()).thenReturn(mockResponseFuture);
+        when(mockResponseFuture.toCompletableFuture()).thenReturn(mockCompletableFuture);
+
+        IndexMetadataConfig metadataConfig = new IndexMetadataConfig();
+        List<String> indexedItems = Arrays.asList("A", "B", "C", "D");
+        metadataConfig.setIndexed(indexedItems);
+        PineconeIndexOperationClient indexDeletionService = new PineconeIndexOperationClient(clientConfig, mockClient);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .indexName("test_name")
+                .dimension(3)
+                .metric("euclidean")
+                .pods(2)
+                .podType("p1.x2")
+                .replicas(2)
+                .metadataConfig(metadataConfig)
+                .sourceCollection("step");
+
+        try{
+            indexDeletionService.createIndex(createIndexRequest);
+        }
+        catch (IOException e) {
+
```

Comment:
Should this be a `fail()` scenario? When would this occur?

Reply:
Did you mean the IOException? Or testing create index if all fields are passed?
If you meant:
1. IOException, then the createIndex method is supposed to throw it, so I've ignored the catching of it. 
2. If creating a new index with all fields test case is required or not, then you're correct it'll not fail right now. In future, I'll like to have more logic included in the validateJsonObject() part of CreateIndexRequest class. I'll remove it for now though.

Reply:
Yeah, it was the IOException I was wondering about since you catch it and then do nothing. If that's intended, some kind of comment would be nice to explain why it's expected.

Reply:
Will update this, thanks.